### ISSUE TITLE
Request help on survival

### ISSUE THREAD

**Issue by glhfgg1024** [CamDavidsonPilon/lifelines#451]
Hi, dear all, thanks a lot for your great tools and the detail documentation! Very helpful!
I have a question: how to interpret the relationship between C-Index and the KM curve? The C-Index is a criterion similar to the classification accuracy, right? And I read several papers, it seems the KM curve is generated from the `\beta` values (if using the CoxPH model), and take the median value (M) as split point, it `beta` is less than `M`, then that sample is belong to the low-risk group, and if `beta` is larger than `M`, then that sample is belong to the high-risk group. I don't understand this. Why could we use this median value to discriminate the high risk and the low risk? I think that depends on the samples themselves.

**Comment by CamDavidsonPilon** [comment]
Hey @glhfgg1024, can you provide some papers that demonstrate this idea? My first thought is to avoid dichotomisation, and treat most outcomes from a "risk" point of view.

**Comment by glhfgg1024** [comment]
Hi, @CamDavidsonPilon , thanks a lot for your kind reply! Please check this paper http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1006076, the last line in page 8, "A log-ranked p-value is then computed to differentiate the Kaplan-Meier survival curves between the higher vs. lower risk groups. In this report, we used the median log hazard ratio as the cutoff threshold."

**Comment by pzivich** [comment]
Haven't had a chance to extensively read through the paper, however it seems like they calculate a prognostic score (risk score) based on some known variables from each of the models (Cox-nnet, standard Cox PH model, CoxBoost, RF-S). I don't know the other techniques really outside of Cox PH but each model should return some probability. Think predicted probabilities of death as a function of chronic conditions, but then add it as a function of time. Based on these calculated scores, they divide it into two categories (high: > 0.5; low: < 0.5). It seems like to me, they wanted to dichotomize  to compare the different models.

I am super confused by what they are doing with the Kaplan Meier. A Cox PH model makes the assumption of proportional hazards, so the generated survival curves will always meet the proportional hazards assumption. The survival curve that comes from the Cox PH is **_NOT_** the same as the survival curves from Kaplan Meier. The Kaplan Meier curve is nonparametric so it does not have to follow the proportional hazards assumption or other distributional assumptions. All a logrank tests compares is the proportional hazards assumption. If a beta is significant in the Cox model, the corresponding logrank test from generated Kaplan Meier curves would also be significant (ignoring conditioning / stratification)

@glhfgg1024 In regards to your thoughts, it is odd to divide them into categories like that. Realistically the probability that comes from the model is more meaningful than the binary category. I would guess it makes their comparisons between models easier, so that's why they did it. Realistically, they could have chosen any point (75th percentile), that why the actual probability generated is more informative. My guess they chose 50th percentile since it maximizes the n for statistical tests in comparing the different models. In practice, clinicians or others would be better informed by the probability itself

**Comment by glhfgg1024** [comment]
Hi @pzivich , thanks a lot for your kind reply and so detail explanation! I'm new on survival prediction. In my opinion, the circumstance is like this: given some patients, we have their clinical outcome and the task of survival prediction is to predict their survival probability in some certain time (e.g., 30 days, 60 days, or 2 months, 6 months). Thus the output of one model (e.g., CoxPH, Cox-Lasso, or many others) is the probability for each patient in each time point. Thus, for each patient, we could get a probability vector that shows the probabilities in each time point of one research period (e.g., 1 month, 2 months, ..., 6 months).  To my limited knowledge of CoxPH, given the features/variables `X` for one patient, the model will compute a `risk` factor by `\beta = Xw`, where `w` is a vector which is learned by minimizing the negative log-likelihood defined by the CoxPH model. For each patient, there is one `\beta`. Thus, for the given set of patients, there are a list of these `\beta`s. Then the authors taken the median of these `\beta`s, and split the set int the `high-risk` group and the `low-risk` group. I'm curious about this. I think the `high-risk` or `low-risk` depends on the patient himself/herself, and should not be from the predicted `\beta`s. I found in many papers, the authors will provide two survival curves or one survival curve as follows:

![a](https://user-images.githubusercontent.com/30264480/39664923-0fa6481c-5051-11e8-81cb-424090dcf2bb.PNG)
![b](https://user-images.githubusercontent.com/30264480/39664926-1536abe6-5051-11e8-89a2-1320abddc368.png)

I think the bottom figure is easy to understand.

**Comment by pzivich** [comment]
@glhfgg1024  no problem. Ahhh okay. I think I see a source of potential confusion. I read the ```beta``` as the model coefficients, not as the sum/one ```beta``` for each patient. If they are using the predicted ```beta``` for each person, dividing into categories based on that value as you described, would seem to be problematic to me. Having said that, this is different and outside of the survival analysis I generally do. I will read through the paper more fully

**Comment by glhfgg1024** [comment]
Hi @pzivich , if I understood correctly, the authors seem to use the predicted median `beta`  as the split point. If you are convenient, please check their code:

https://github.com/traversc/cox-nnet/blob/8e73a1145f7a6c1577f8a107120b0494571e8541/examples/plots.r#L6

In my opinion with limited knowledge of survival prediction, we are given some patients (say 60), according to their clinical diagnosis, we first split them into two groups of `high-risk` or `low-risk`, then we split them into two sets (`train` and `validation`), then we train the model using the `train` set to obtain the `beta` coefficient for CoxPH, and using the `validation` set to do cross validation to select the hyper-parameters (e.g., the regularization parameters for L1/L2 regularization energy on `\beta`). Then, we are given the `test` set, we use the obtained model to predict their survival probabilities.
